.github/workflows/build.yaml

```
@@ -0,0 +1,89 @@
+name: Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: ["contribute"]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        runner: [windows-viewer]
+        configuration: [ReleaseOS]
+        addrsize: [64]
+        include:
+          - runner: windows-viewer
+            configuration: ReleaseOS
+            addrsize: 32
+    runs-on: ${{ matrix.runner }}
+    env:
+      AUTOBUILD_CONFIGURATION: ${{ matrix.configuration }}
+      AUTOBUILD_ADDRSIZE: ${{ matrix.addrsize }}
+      AUTOBUILD_INSTALLABLE_CACHE: ${{ github.workspace }}/.autobuild-installables
+      AUTOBUILD_VARIABLES_FILE: ${{ github.workspace }}/.build-variables/variables
+      AUTOBUILD_VSVER: "170" # vs2k22
+      LOGFAIL: debug # Show details when tests fail
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

Similar to as we do elsewhere, should we do:
```suggestion
        uses: actions/checkout@v3
        with:
          ref: ${{ github.event.pull_request.head.sha || github.sha }}
```
Otherwise, for builds that are the result of a PR event, the `ref` will point to the merge commit created by GitHub, which could be undesirable for our builds.